lib/mirage/mirage.mli

```diff
@@ -764,6 +764,114 @@ val default_argv : argv impl
 val no_argv : argv impl
 (** [no_argv] Disable command line parsing and set argv to [|""|]. *)
 
+(** {2 Git client configuration} *)
+
+(** The user is able to initiate a connection with a Git repository. However,
+    there are many ways to communicate with a Git repository:
+
+    - TCP/IP
+    - HTTP
+    - HTTP + TLS
+    - SSH
+
+    The devices defined below define these composable ways. The
+    [git_client impl] returned from them can be passed to Git or Irmin in order
+    to be able to {i fetch} and {i push} from/into a Git repository.
+
+    The user is able to restrict or enlarge protocol possibilities needed for
+    its application. For instance, the user is able to restrict only the SSH
+    connection to communicate with a Git repository or the user can handle
+    TCP/IP and SSH as possible protocols to communicate with a peer.
+
+    For instance, a device which is able to communicate {i via} TCP/IP and SSH
+    can be implemented like:
+
+    {[
+      let git_client =
+        let dns = git_happy_eyeballs stackv4v6 in
+        let ssh = git_ssh ~key (git_tcpv4v6_of_stackv4v6 stackv4v6) dns in
+        let tcp = git_tcp (git_tcpv4v6_of_stackv4v6 stackv4v6) dns in
+        git_merge_clients ssh tcp
+    ]} *)
+
+type git_client
+(** Type of a device which implements a {i certain} protocol. *)
```

Comment:
`certain` is not clear here: I guess you mean "Git"?

Reply:
The best will be: the Git protocol over a _certain_ transmission protocol. WDYT?

Reply:
I'd go with a simpler form as `certain` doesn't mean the same thing in French and English so it's confusing here.

```suggestion
(** The type for devices that implement the Git protocol. *)
```